### ISSUE TITLE
Test x-amz-mp-parts-count header

### ISSUE THREAD

**Issue by gaul** [ceph/s3-tests#214]
Amazon returns `x-amz-mp-parts-count` in response to a get object metadata request.  This can be used to find the number of parts of a multipart object:

https://github.com/aws/aws-sdk-java/blob/4734de6fb0f80fe5768a6587aad3b9d0eaec388f/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectMetadataRequest.java#L387-L397

**Comment by jamshid** [comment]
Yes this would be a good test because the AWS Java SDK TransferManager uses this undocumented header, though they are talking about changing their code to not rely on it: 
https://github.com/aws/aws-sdk-java/issues/1303
